src/usage_metrics/core/kaggle.py

```diff
@@ -0,0 +1,71 @@
+"""Transform data from Kaggle logs."""
+
+import os
+
+import pandas as pd
+from dagster import (
+    AssetExecutionContext,
+    WeeklyPartitionsDefinition,
+    asset,
+)
+
+
+@asset(
+    partitions_def=WeeklyPartitionsDefinition(start_date="2023-08-16"),
+    io_manager_key="database_manager",
+    tags={"source": "kaggle"},
+)
+def core_kaggle_logs(
+    context: AssetExecutionContext,
+    raw_kaggle_logs: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform daily S3 logs.
+
+    Add column headers, geocode values,
+    """
+    context.log.info(f"Processing data for the week of {context.partition_key}")
+
+    if raw_kaggle_logs.empty:
+        context.log.warn(f"No data found for the week of {context.partition_key}")
+        return raw_kaggle_logs
+
+    # Where there is a "nullable" column, drop the corresponding "has" and duplicate column from data
+    nullable_cols = [col for col in raw_kaggle_logs.columns if "Nullable" in col]
+    duplicated_cols = [col.replace("Nullable", "") for col in nullable_cols] + [
+        "has{}".format((col[0].upper() + col[1:]).replace("Nullable", ""))
+        for col in nullable_cols
+    ]
```

Comment:
Could you explain a little more what the raw data looks like and what you're trying to fix here?

Reply:
Added a description into the transform method documentation.

Reply:
thank you!